pkg/types/baremetal/platform.go

```diff
@@ -19,6 +19,7 @@ type Host struct {
 	Role            string `json:"role"`
 	BootMACAddress  string `json:"bootMACAddress" validate:"required,uniqueField"`
 	HardwareProfile string `json:"hardwareProfile"`
+	RootHint        string `json:"rootHint"`
```

Comment:
I know it's still under discussion but if we'll bring root device hints out to the user, it should be something like we do with BMC with a type, there's no reason to do JSON and especially not JSON in a string.  We don't want to be taking unvalidatable, arbitrary data.

Reply:
New version removes the json as a string, so it should remove the worry of having data that wouldn't unmarshal. I'm a little reluctant to start validating the data, there are about 10 potential fields that could be given, and the values can themselves contain opperators. The validation could get complex, would be duplicated in ironic code and would have to be kept in sync.

https://docs.openstack.org/ironic/latest/install/advanced.html#specifying-the-disk-for-deployment-root-device-hints

Reply:
The [upstream design](https://github.com/metal3-io/metal3-docs/blob/master/design/user-defined-root-device-hints.md) in metal3 is going to use separate fields. I think we're just going to end up passing the values through to ironic as they're given to us. Hints can be combined, so we need to allow all of the fields to be present. It would be good of the structure used in the installer input matched what the BareMetalHost will be using.